include/ck/host_utility/device_prop.hpp

```diff
@@ -9,25 +9,12 @@
 
 namespace ck {
 
-inline std::string get_device_name()
+// https://github.com/ROCm/MIOpen/blob/8498875aef84878e04c1eabefdf6571514891086/src/target_properties.cpp#L40
+// broken out and guarded with function call to avoid issues with initialization and
+// dynamic linking that can corrupt the map's contents
+inline const std::map<std::string, std::string>& get_device_name_map()
 {
-    hipDeviceProp_t props{};
-    int device;
-    auto status = hipGetDevice(&device);
-    if(status != hipSuccess)
-    {
-        return std::string();
-    }
-
-    status = hipGetDeviceProperties(&props, device);
-    if(status != hipSuccess)
-    {
-        return std::string();
-    }
-    const std::string raw_name(props.gcnArchName);
-
-    // https://github.com/ROCm/MIOpen/blob/8498875aef84878e04c1eabefdf6571514891086/src/target_properties.cpp#L40
-    static std::map<std::string, std::string> device_name_map = {
+    static const std::map<std::string, std::string> device_name_map = {
```

Comment:
Hmm I'd first ask a question do we really need this map in here?? Its content is pretty archaic... Maybe we could relay only on device prop instead?